test/e2e/proxy_publicviewer_test.go

```diff
@@ -0,0 +1,289 @@
+	default:
+		return ""
+	}
+}
+
+// tests access to community-shared spaces
+func TestProxyPublicViewer(t *testing.T) {
+	// make sure everything is ready before running the actual tests
+	awaitilities := WaitForDeployments(t)
+	hostAwait := awaitilities.Host()
+	memberAwait := awaitilities.Member1()
+
+	// public viewer is enabled in ToolchainConfig
+	hostAwait.UpdateToolchainConfig(t, testconfig.PublicViewerConfig(true))
+
+	// we create a space to share
+	space, _, _ := testsupportspace.CreateSpace(t, awaitilities,
+		testspace.WithTierName("appstudio"),
+		testspace.WithSpecTargetCluster(memberAwait.ClusterName),
+	)
+
+	// users test cases
+	tt := map[string]struct {
+		proxyClientUser func() proxyUser
+	}{
+		"approved user with space": {
+			proxyClientUser: func() proxyUser {
+				user := createAppStudioCommunityUser(t, awaitilities, func(sr *SignupRequest) *SignupRequest {
+					return sr.
+						ManuallyApprove().
+						EnsureMUR().
+						RequireConditions(
+							wait.ConditionSet(wait.Default(), wait.ApprovedByAdmin())...)
+				})
+				return proxyUser{signupResult: user}
+			},
+		},
+		"approved user without space": {
+			proxyClientUser: func() proxyUser {
+				user := createAppStudioCommunityUser(t, awaitilities, func(sr *SignupRequest) *SignupRequest {
+					return sr.
+						ManuallyApprove().
+						EnsureMUR().
+						NoSpace().
+						RequireConditions(
+							wait.ConditionSet(wait.Default(), wait.ApprovedByAdmin())...)
+				})
+				return proxyUser{signupResult: user}
+			},
+		},
+		"not approved user": {
+			proxyClientUser: func() proxyUser {
+				user := createAppStudioCommunityUser(t, awaitilities, func(sr *SignupRequest) *SignupRequest {
+					return sr.
+						NoSpace().
+						RequireConditions(
+							wait.ConditionSet(wait.Default(), wait.VerificationRequired())...)
+				})
+				return proxyUser{signupResult: user}
+			},
+		},
+		"sso user": {
+			proxyClientUser: func() proxyUser {
+				userIdentity := &commonauth.Identity{
+					ID:       uuid.New(),
+					Username: "joe",
+				}
+				email := "joe@joe.joe"
+				claims := []commonauth.ExtraClaim{commonauth.WithEmailClaim(email)}
+				token, err := authsupport.NewTokenFromIdentity(userIdentity, claims...)
+				require.NoError(t, err)
+				return proxyUser{token: token, email: email}
+			},
+		},
+	}
+
+	t.Run("space is flagged as community", func(t *testing.T) {
+		sb := CreateCommunitySpaceBinding(t, hostAwait, space.Name, space.Namespace)
+		require.NotNil(t, sb)
+
+		// Wait until space is flagged as community
+		_, err := hostAwait.WaitForSpaceBinding(t, toolchainv1alpha1.KubesawAuthenticatedUsername, space.Name, wait.UntilSpaceBindingHasSpaceRole("viewer"))
+		require.NoError(t, err)
+
+		// wait until the space has ProvisionedNamespaces
+		sp, err := hostAwait.WaitForSpace(t, space.Name, wait.UntilSpaceHasAnyProvisionedNamespaces())
+		require.NoError(t, err)
+
+		t.Run("user is not banned", func(t *testing.T) {
```

Comment:
TBH I'm not sure we should have the subtests for "user is not banned" and "user is banned", since the not approved user and the sso user cannot be banned, or at least I don't think we ban those right now, it's not a real scenario. 

I would maybe just add an entry for the banned user in the `tt` ( testing table ) and verify that it cannot access community spaces.

Reply:
With the current implementation of the [ensureUserIsNotBanned Middleware](https://github.com/codeready-toolchain/registration-service/blob/e15d7cc1730b9b07523bd5c91d89250634d360cb/pkg/proxy/proxy.go#L412-L443)  they can be banned. Indeed, the current implementation checks if a `BannedUser` with the same mail it finds in the JWT token exists. This means that we can technically ban not approved and SSO users.

We are not making use of this scenario ATM, but it's supported by the current implementation.

Reply:
ok makes sense. Thanks for the explanation. Let's keep them there.